lib/diff.ml

```diff
@@ -29,7 +29,7 @@ type ('same, 'diff) option_ = ('same option, ('same, 'diff) entry) maybe_changed
 type 'same atomic_option = ('same, 'same atomic_modification) option_
 
 type ('same, 'diff) atomic_variant =
-  ('same, ('same atomic_modification, 'diff) Either.t) maybe_changed
+  ('same, ('same atomic_modification, 'diff) maybe_changed) maybe_changed
```

Comment:
My suggestion was "Change means changed variant and Same means same variant" so you would need to invert the type params here:
```ocaml
type ('same, 'diff) atomic_variant =
  ('same, ('diff, 'same atomic_modification) maybe_changed) maybe_changed
```
however, I think Nathan suggested to define another variant here:
```
type ('same_variant, 'different_variant) variant_change =
  | Same_variant of 'same_variant
  | Different_variant of 'different_variant
```

(can you confirm that was your suggestion @NathanReb ?)

Reply:
Yes I think that would be better than to hide a diff in the "no diff" side of a `maybe_changed`!